src/entity_factuality.py

```diff
@@ -101,16 +95,33 @@ def classify_entities(
         sources_by_id,
     ) -> MarkedEntityLookup:
         classified_entities: MarkedEntityLookup = {}
+        # Build a list of (sum, source, ents[]) to enable
+        # batching across summaries when extracting feaatures
+        ents_to_classify: InferenceInput = []
         for sum_id, summary in gen_summaries_by_id.items():
             updated_entities = [x.copy() for x in marked_entities[sum_id]]
-            entities_to_classify = [x for x in updated_entities if not x["in_source"]]
-            if len(entities_to_classify) > 0:
-                features = self.extract_features(
-                    summary, sources_by_id[sum_id], entities_to_classify
-                )
-                predictions = self.clf.predict(features)
-                for entitiy, pred in zip(entities_to_classify, predictions):
-                    entitiy["predicted_label"] = self.label_mapping[pred]
+            for ent in updated_entities:
+                if ent["in_source"]:
+                    ent["predicted_label"] = ANNOTATION_LABELS["Non-hallucinated"]
 
+            ents_without_prediction = [ent for ent in updated_entities if not ent["in_source"]]
```

Comment:
```suggestion
            ents_not_in_source = [ent for ent in updated_entities if not ent["in_source"]]
```

current name a bit confusing considering that, at the data level, all entities have a `predicted_label`, right?